milli/src/search/new/exact_attribute.rs

```diff
@@ -0,0 +1,253 @@
+        let mut candidates = universe.clone();
+        let words_positions: Vec<(Vec<_>, _)> = exact_term_position_ids
+            .iter()
+            .copied()
+            .map(|(term, position, _)| (term.interned_words(ctx).collect(), position))
+            .collect();
+        for (words, position) in &words_positions {
+            if candidates.is_empty() {
+                return Ok(State::Empty(query_graph.clone()));
+            }
+
+            'words: for (offset, word) in words.iter().enumerate() {
+                let offset = offset as u16;
+                let word = if let Some(word) = word {
+                    word
+                } else {
+                    continue 'words;
+                };
+                // Note: Since the position is stored bucketed in word_position_docids, for queries with a lot of
+                // longer phrases we'll be losing on precision here.
+                let bucketed_position = crate::bucketed_position(position + offset);
+                let word_position_docids = CboRoaringBitmapCodec::bytes_decode(
+                    ctx.get_db_word_position_docids(*word, bucketed_position)?.unwrap_or_default(),
+                )
+                .unwrap_or_default();
+                candidates &= word_position_docids;
+            }
+        }
+
+        let candidates = candidates;
+
+        if candidates.is_empty() {
+            return Ok(State::Empty(query_graph.clone()));
+        }
+
+        let searchable_fields_ids = ctx.index.searchable_fields_ids(ctx.txn)?.unwrap_or_default();
+
+        let mut candidates_per_attribute = Vec::with_capacity(searchable_fields_ids.len());
+
+        // then check that there exists at least one attribute that has all of the terms
+        for fid in searchable_fields_ids {
+            let mut intersection = MultiOps::intersection(
+                words_positions
+                    .iter()
+                    .flat_map(|(words, ..)| words.iter())
+                    // ignore stop words words in phrases
+                    .flatten()
+                    .map(|word| -> Result<_> {
+                        Ok(ctx
+                            .get_db_word_fid_docids(*word, fid)?
+                            .map(CboRoaringBitmapCodec::bytes_decode)
+                            .unwrap_or_default()
+                            .unwrap_or_default())
+                    }),
+            )?;
+            intersection &= &candidates;
+            if !intersection.is_empty() {
+                let candidates_with_exact_word_count = ctx
+                    .index
+                    .field_id_word_count_docids
+                    .get(ctx.txn, &(fid, exact_term_position_ids.len() as u8))?
+                    .unwrap_or_default();
+                candidates_per_attribute.push(FieldCandidates {
+                    start_with_exact: intersection,
+                    exact_word_count: candidates_with_exact_word_count,
+                });
```

Comment:
do we have to store them all in an array, or is there a way to perform the union here already?